src/Stratis.Bitcoin.Tests/P2P/PeerConnectorTests.cs

```diff
@@ -210,6 +210,141 @@ public void PeerConnectorDiscovery_ConnectsTo_DiscoveredPeers()
+            var endpointPeer2 = new IPEndPoint(addressInDifferentNetworkGroupPeer2, 80);
+            connectedToThisPeer = ConnectToPeer(peerAddressManager, networkPeerFactoryExisting, connectionManagerSettingsExisting, peerConnector, endpointPeer2);
+            Assert.True(connectedToThisPeer);
+
+            // Peer 3 is a local address in a different group.
+            IPAddress addressInDifferentNetworkGroupPeer3 = IPAddress.Parse("0:0:0:0:0:ffff:c0a8:101"); // ipv6
+            var endpointPeer3 = new IPEndPoint(addressInDifferentNetworkGroupPeer3, 80);
+            connectedToThisPeer = ConnectToPeer(peerAddressManager, networkPeerFactoryExisting, connectionManagerSettingsExisting, peerConnector, endpointPeer3);
+            Assert.True(connectedToThisPeer);
+        }
+
+        [Fact]
+        public void PeerConnectorDiscovery_DoesNotConnectTo_NodeInSameNetworkGroup_WithIpRangeFilteringEnabled()
+        {
+            // Setup.
+            Network mainNetwork = Network.Main;
+            DataFolder peerFolder = CreateDataFolder(this);
+            var peerAddressManager = new PeerAddressManager(DateTimeProvider.Default, peerFolder, this.extendedLoggerFactory, new SelfEndpointTracker());
+            var nodeSettings = new NodeSettings(mainNetwork, args: new[] { "-IpRangeFiltering" });
+
+            var connectionManagerSettingsExisting = new ConnectionManagerSettings(nodeSettings);
+            Mock<INetworkPeerFactory> networkPeerFactoryExisting = new Mock<INetworkPeerFactory>();
+
+            var peerConnector = new PeerConnectorConnectNode(this.asyncLoopFactory, DateTimeProvider.Default, this.extendedLoggerFactory, this.network, networkPeerFactoryExisting.Object, this.nodeLifetime, nodeSettings, connectionManagerSettingsExisting, peerAddressManager, new SelfEndpointTracker());
+
+            IConnectionManager connectionManagerExisting = this.CreateConnectionManager(nodeSettings, connectionManagerSettingsExisting, peerAddressManager, peerConnector);
+            peerConnector.Initialize(connectionManagerExisting);
+
+            // Original peer.
+            IPAddress originalAddressPeer1 = IPAddress.Parse("::ffff:213.86.221.35"); // ipv4
+            var endpointPeer1 = new IPEndPoint(originalAddressPeer1, 80);
+            bool connectToThisPeer = ConnectToPeer(peerAddressManager, networkPeerFactoryExisting, connectionManagerSettingsExisting, peerConnector, endpointPeer1);
+            Assert.True(connectToThisPeer);
+            
+            // Peer from same group.
+            IPAddress addressInSameNetworkGroupPeer2 = IPAddress.Parse("::ffff:213.86.221.36"); // ipv4
+            var endpointPeer2 = new IPEndPoint(addressInSameNetworkGroupPeer2, 80);
+            connectToThisPeer = ConnectToPeer(peerAddressManager, networkPeerFactoryExisting, connectionManagerSettingsExisting, peerConnector, endpointPeer2);
+            Assert.False(connectToThisPeer);
+
+            // Peer from same group: (213.86.221.37)
```

Comment:
Wrong comment?

Also why this peer should fail to connect?

Reply:
Have corrected and standardized tests so they use same data, same positive and negative tests and same mix of ipv4 and ipv6.